test-result-summary-client/src/TabularView/TabularView.jsx

```diff
@@ -0,0 +1,703 @@
+                this.state[key] = value;
+            }
+        }
+        this.initializeJdk();
+        this.showData('test');
+        this.showData('baseline');
+     }
+
+    // Set default values for choosing the JDK including date, version, type and sdk resource if URL has missing information
+    initializeJdk () {
+        const date = (new Date().getDate()).toString();
+        const month = (new Date().getMonth() + 1).toString(); //Current Month
+        const year = (new Date().getFullYear()).toString(); //Current Year
+
+        const jdkDate = year + ((month.length < 2) ? "0" + month : month) + ((date.length < 2) ? "0" + date : date)
+        
+        !('baselineJdkDate' in this.state) && (this.state.baselineJdkDate = jdkDate);
+        !('baselineJdkVersion' in this.state) && (this.state.baselineJdkVersion = 'O8');
+        !('baselineJvmType' in this.state) && (this.state.baselineJvmType = 'j9');
+        !('baselineSdkResource' in this.state) && (this.state.baselineSdkResource = 'releases');
+
+        !('testJdkDate' in this.state) && (this.state.testJdkDate = jdkDate);
+        !('testJdkVersion' in this.state) && (this.state.testJdkVersion = 'O8');
+        !('testJvmType' in this.state) && (this.state.testJvmType = 'j9');
+        !('testSdkResource' in this.state) && (this.state.testSdkResource = 'nightly');
+    }
+    // Handle simple event changes for JDK version, type and sdk resource
+    handleChange(event) {
+        console.log("Handle Change: ", event.target);
+        this.setState({
+            [event.target.name]: event.target.value
+        });
+    }
+    // Set the color state and call the filter
+    handleColorFilter (event) {
+        this.setState({
+            colorFilter: event.target.value
+        }, () => {
+        	// Color changed, signal to call benchmark filter first
+            this.colorFilter(true);
+        });
+    }
+ 
+    handleDayChange(selectedDay, modifiers, dayPickerInput) {
+
+        const input = dayPickerInput.getInput();
+        this.setState({
+            selectedDay,
+            isEmpty: !input.value.trim(),
+            isDisabled: modifiers.disabled === true,
+        }, function () {
+            if (this.state.selectedDay !== undefined) {
+                // Transform date to correct format
+                this.dateTransform(this.state.selectedDay.toLocaleDateString(), dayPickerInput.props.dayPickerProps.type);}
+        });
+    }
+
+    setTreeData() {
+    	// Set the tree select values for benchmark filter
+        let newArray = [];
+        this.state.testData.forEach(function (elem) {
+        let testEntry = elem;
+        let benchmark = testEntry.benchmarkName.split(",")[0];
+        let variant = testEntry.benchmarkName.split(",")[1];
+        let metric = testEntry.benchmarkName.split(",")[2];
+
+        let topLevel = {};
+        let midLevel = {};
+        let child = {title: metric, value: testEntry.benchmarkName};
```

Comment:
We should mention that we are giving unique titles for these values in order to avoid showing metrics for all benchmark variants. By setting titles, we limit the display to one variant, requiring user to manually select different variants in case one wants to look up the metric for multiple variants. For example, footprint metric exists in multiple Liberty variants such as DT7, DT3 and AcmeAir.